src/core/Akka.Cluster/Member.cs

```diff
@@ -92,6 +93,20 @@ internal Member(UniqueAddress uniqueAddress, int upNumber, MemberStatus status,
             Roles = roles;
         }
 
+        /// <summary>
+        /// Used when `akka.actor.serialize-messages = on`.
+        /// </summary>
+        /// <param name="uniqueAddress">The address of the member.</param>
+        /// <param name="upNumber">The upNumber of the member, as assigned by the leader at the time the node joined the cluster.</param>
+        /// <param name="status">The status of this member.</param>
+        /// <param name="roles">The roles for this member. Can be empty.</param>
+        [JsonConstructor]
+        internal Member(UniqueAddress uniqueAddress, int upNumber, MemberStatus status, IEnumerable<string> roles)
```

Comment:
The issue with serializing the `Member` class is an old, but familiar one: JSON.NET is unable to serialize the `System.Collections.Immutable.ImmutableHashSet<string>` taken for the `roles` parameter on the normal constructor this actor uses. Thus, I created a second `internal` constructor that just takes an `IEnumerable<string>` instead and marked that as the `JsonConstructor` - ideally once Hyperion is done we will no longer need this.